### PR TITLE
[Java] Update dependencies

### PR DESCRIPTION
**Pull request by ackintosh**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Updated Retrofit2 to address the vulnerabilities:

- https://nvd.nist.gov/vuln/detail/CVE-2018-1000850
- https://nvd.nist.gov/vuln/detail/CVE-2018-1000844

### CODE REVIEW

**Comment by ackintosh** [comment]
https://circleci.com/gh/OpenAPITools/openapi-generator/3871#tests/containers/1

```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2-play25/src/main/java/org/openapitools/client/Play25CallFactory.java:[92,12] org.openapitools.client.Play25CallFactory.PlayWSCall is not abstract and does not override abstract method timeout() in okhttp3.Call
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project petstore-java-client-retrofit2-play25: Compilation failure
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2-play25/src/main/java/org/openapitools/client/Play25CallFactory.java:[92,12] org.openapitools.client.Play25CallFactory.PlayWSCall is not abstract and does not override abstract method timeout() in okhttp3.Call
```

**Comment by ackintosh** [comment]
Related PR @ okhttp: https://github.com/square/okhttp/pull/4363

**Comment by ackintosh** [comment]
cc @bbdouglas (2017/07) @JFCote (2017/08) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)

File: modules/openapi-generator/src/main/resources/Java/libraries/retrofit2/play25/Play25CallFactory.mustache [hunk 1]
```diff
@@ -96,18 +98,25 @@ public class Play25CallFactory implements okhttp3.Call.Factory {
...
+            this.timeout = new AsyncTimeout();
         }
 
         @Override
         public Request request() {
             return request;
         }
 
+        @Override
+            public Timeout timeout() {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/retrofit2/play25/Play25CallFactory.mustache]: @ackintosh looks like it's not indented properly (4 extra spaces)

**Comment by ackintosh** [comment]
Thanks for the review! Fixed the indent.

**Pull request merged**